### ISSUE TITLE
Flickering between aliased and non-aliased graphics on macOS

### ISSUE THREAD

**Issue by asbjornu** [schismtracker/schismtracker#334]
On macOS Big Sur 11.7, Schism Tracker version `20220905` (perhaps also earlier; I haven't tested) through `20221020` has a strange flicker between what seems to be aliased and non-aliased graphics:

![Schism Tracker flickering](https://user-images.githubusercontent.com/12283/204540577-2e4c8c7c-2aa8-48a1-b2da-17366d305566.gif)

This only seems to occur when a module is not playing, though. When playing a module, the UI stays in the non-aliased mode until I hit F8 and the flickering starts again. The flickering seems to occur on all screens.

**Comment by jangler** [comment]
Does using different video scaling options on the Ctrl+F1 screen affect this at all?

**Comment by asbjornu** [comment]
Changing video scaling does not affect the flickering, but running it in fullscreen does. In fullscreen mode, all video scaling modes behave nicely.

**Comment by jangler** [comment]
There were some changes to macOS integration and video interpolation in today's release, can you test again?

**Comment by asbjornu** [comment]
Version `20221201` is still flickering, I'm afraid.

**Comment by mrpapersonic** [comment]
Does this still happen with the latest release?

**Comment by asbjornu** [comment]
Yes, but the flicker seems to be less consistent. Here's a screen capture of version `20240129` taken with macOS 13.6.4 on a 16" MacBook Pro M2.

![Schism Tracker flickering](https://github.com/schismtracker/schismtracker/assets/12283/033474bd-9f5d-4095-a666-a081c4153e23)

**Comment by mrpapersonic** [comment]
I have a feeling this has something do with retina/high dpi displays. I'll see if I can replicate it on Windows.

**Comment by mrpapersonic** [comment]
I did a bit of refactoring and removed some unnecessary bits of the video code recently; if you could test it again I'd appreciate it since I don't have a real mac anywhere.

Adding on my previous comment, I couldn't replicate this on Windows even using the "retina" 125% mode.

**Comment by asbjornu** [comment]
@mrpapersonic no change in version 20240426, unfortunately. The flickering remains.

**Comment by abandpass** [comment]
I am able to reproduce this on my macbook air M2.  However when using the same build on my M1 mac mini connected to 2x 1080p screens, all is well.

Both machines are running mac OS 15.0.1.  It occurs somewhat randomly, more in the info screen but rarely in the pattern screen.

**Comment by asbjornu** [comment]
> I am able to reproduce this on my macbook air M2. However when using the same build on my M1 mac mini connected to 2x 1080p screens, all is well.

I just tested with an external 4k monitor on an MacBook Pro M2, and as far as I can see, the problem isn't visible there. So this might indeed only be a problem on the native Mac monitors with retina resolution.

**Comment by mrpapersonic** [comment]
I honestly have no idea whats happening here. I've tried using high DPI on other systems and it works perfectly fine.

It *could* be possible that the lack of vsync is causing issues, but I doubt it would cause anything like this.

**Comment by mrpapersonic** [comment]
> I am able to reproduce this on my macbook air M2. However when using the same build on my M1 mac mini connected to 2x 1080p screens, all is well.

Definitely a high-DPI issue then.

Okay, looking back at this -- would a kind soul dig into `Schism Tracker.app/Contents/Info.plist` and change
```
	<key>NSHighResolutionCapable</key>
	<false/>
```
to
```
	<key>NSHighResolutionCapable</key>
	<true/>
```
and see if that makes any difference?

**Comment by asbjornu** [comment]
@mrpapersonic Yep, that seems to fix it! At least according to my limited test on a Retina 5K, 27", 2020 iMac with macOS 14.7.7.
